diamond-api/src/find.rs

```diff
@@ -85,20 +85,29 @@ pub struct FindQuery {
 
 impl FromRequest for FindQuery {
     type Error = Error;
-    type Future = Box<dyn Future<Item = Self, Error = Error>>;
+    type Future = LocalBoxFuture<'static, Result<Self, Self::Error>>;
     type Config = ();
 
     fn from_request(req: &HttpRequest, payload: &mut dev::Payload) -> Self::Future {
         match req.content_type().to_lowercase().as_str() {
-            "application/x-www-form-urlencoded" => {
-                Box::new(Form::<FindQuery>::from_request(req, payload).map(|x| x.into_inner()))
-            }
-            "application/json" => {
-                Box::new(Json::<FindQuery>::from_request(req, payload).map(|x| x.into_inner()))
-            }
-            _ => Box::new(result(
-                Query::<FindQuery>::from_request(req, payload).map(|x| x.into_inner()),
-            )),
+            "application/x-www-form-urlencoded" => Form::<FindQuery>::from_request(req, payload)
+                .map(|r| match r {
+                    Ok(v) => Ok(v.into_inner()),
+                    Err(e) => Err(e.into()),
+                })
```

Comment:
`|r| Ok(r?.into_inner())`

Reply:
This is just a suggestion. Feel free to ignoew

Reply:
good catch.